From b0f01086e440516c03ce61e5597b9a0d2df0a1c0 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Wed, 1 Nov 2017 16:17:43 +0000
Subject: [PATCH 333/454] bcm2835-unicam: Revert changes for notifier fn ptrs

4.15 is using const function pointers for the notifier
functions. Those changes don't cherry-pick easily, so
revert those mods in this driver.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/media/platform/bcm2835/bcm2835-unicam.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

--- a/drivers/media/platform/bcm2835/bcm2835-unicam.c
+++ b/drivers/media/platform/bcm2835/bcm2835-unicam.c
@@ -1833,11 +1833,6 @@ static int unicam_async_complete(struct
 	return unicam_probe_complete(unicam);
 }
 
-static const struct v4l2_async_notifier_operations unicam_async_ops = {
-	.bound = unicam_async_bound,
-	.complete = unicam_async_complete,
-};
-
 static int of_unicam_connect_subdevs(struct unicam_device *dev)
 {
 	struct platform_device *pdev = dev->pdev;
@@ -1956,7 +1951,8 @@ static int of_unicam_connect_subdevs(str
 	subdevs[0] = asd;
 	dev->notifier.subdevs = subdevs;
 	dev->notifier.num_subdevs = 1;
-	dev->notifier.ops = &unicam_async_ops;
+	dev->notifier.bound = unicam_async_bound;
+	dev->notifier.complete = unicam_async_complete;
 	ret = v4l2_async_notifier_register(&dev->v4l2_dev,
 					   &dev->notifier);
 	if (ret) {
